examples/rust/service_attributes/creator.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use core::time::Duration;
+use iceoryx2::prelude::*;
+
+const CYCLE_TIME: Duration = Duration::from_secs(1);
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let service_name = ServiceName::new("Service/With/Properties")?;
+
+    let service = zero_copy::Service::new(&service_name)
+        .publish_subscribe::<u64>()
+        .create_with_attributes(
+            // define a set of properties that are static for the lifetime
+            // of the service
+            &DefinedAttributes::new()
+                .define("dds_service_mapping", "my_funky_service_name")
+                .define("tcp_serialization_format", "cdr")
+                .define("someip_service_mapping", "1/2/3")
+                .define("camera_resolution", "1920x1080"),
```

Comment:
I wonder whether it would make sense to have custom attributes and buildin attributes, e.g.
```
// vendor-group-key
iox2-camera-resolution
iox2-network-protocol
...
foo-bar-attribute
```

Browser use the same mechanism for CSS extensions which are not yet in the standard

Reply:
We could reserve the prefix `iox2` for any key value. However, trying to define a specific list effectively tends to result in an infinitely long list with many never-used key values.

Reply:
I don't understand what you mean :sweat_smile: